ServerCore/Pages/Submissions/Index.cshtml.cs

```diff
@@ -370,18 +370,18 @@ orderby submission.TimeSubmitted
             foreach (SubmissionView submissionView in SubmissionViews)
             {
                 Submissions.Add(submissionView.Submission);
-            }
 
-            if (PuzzleState.SolvedTime != null)
-            {
-                if (!Puzzle.IsFreeform && Submissions?.Count > 0)
+                if (submissionView.Response != null
+                    && submissionView.Response.IsSolution 
+                    && !Puzzle.IsFreeform)
                 {
                     AnswerToken = Submissions.Last().SubmissionText;

```

Comment:
Why is this not submissionView.Submission.SubmissionText?

Reply:
AH. Good point